### PR TITLE
Added rewards to ProfileStorage for operator fees collection

### PR DESCRIPTION
**Pull request by u-hubar**

### CODE REVIEW

File: contracts/Profile.sol [hunk 1]
```diff
@@ -92,6 +107,46 @@ contract Profile {
...
+
+        UnorderedIndexableContractDynamicSetLib.Contract[] memory hashFunctions = hp.getAllHashFunctions();
+        uint256 hashFunctionsNumber = hashFunctions.length;
+        for (uint8 i; i < hashFunctionsNumber; ) {
+            ps.setNodeAddress(identityId, hashFunctions[i].id);
+            unchecked { i++; }
+        }
+    }
+
+    function stakeReward(uint72 identityId) external onlyAdmin(identityId) {
```
Comment (branarakic) [inline contracts/Profile.sol:123]: lets rename the function to stakeAccumulatedOpFee (so its clearer). technically these are not rewards

File: contracts/Profile.sol [hunk 2]
```diff
@@ -92,6 +107,46 @@ contract Profile {
...
+
+        ps.setReward(identityId, 0);
+        stakingContract.addStake(identityId, rewardAmount);
+
+        if (!shardingTableStorage.nodeExists(identityId) && newStake >= parametersStorage.minimumStake()) {
+            shardingTable.pushBack(identityId);
+        }
+    }
+
+    function withdrawReward(uint72 identityId) external onlyAdmin(identityId) {
```
Comment (branarakic) [inline contracts/Profile.sol:140]: withdrawAccumulatedOpFee

File: contracts/storage/ProfileStorage.sol [hunk 3]
```diff
@@ -93,6 +90,14 @@ contract ProfileStorage {
         emit AskUpdated(identityId, ask);
     }
 
+    function getReward(uint72 identityId) external view returns (uint96) {
+        return profiles[identityId].reward;
```
Comment (branarakic) [inline contracts/storage/ProfileStorage.sol:94]: general comment on wording - "reward" sounds problematic.
We should stick with "fees" (accumulated fees, op fees etc) for node operator fees, and have "reward" be used for the rest (tokens accumulating in stake)

Comment (branarakic) [reply contracts/storage/ProfileStorage.sol:94]: also makes reviewing the code harder

File: contracts/Profile.sol [hunk 4]
```diff
@@ -92,6 +107,46 @@ contract Profile {
...
+        uint96 rewardAmount = ps.getReward(identityId);
+        require(rewardAmount != 0, "You have no reward");
+
+        uint96 oldStake = stakingStorage.totalStakes(identityId);
+        uint96 newStake = oldStake + rewardAmount;
+
+        ps.setReward(identityId, 0);
+        stakingContract.addStake(identityId, rewardAmount);
+
+        if (!shardingTableStorage.nodeExists(identityId) && newStake >= parametersStorage.minimumStake()) {
```
Comment (branarakic) [inline contracts/Profile.sol:135]: is this necessary? I believe the addStake function will handle this part

**Review by branarakic — CHANGES_REQUESTED** [review]

File: contracts/Profile.sol (hunk 4 above)
Comment (u-hubar) [reply contracts/Profile.sol:135]: Yes, my bad

File: contracts/Profile.sol (hunk 1 above)
Comment (u-hubar) [reply contracts/Profile.sol:123]: Done

File: contracts/Profile.sol (hunk 2 above)
Comment (u-hubar) [reply contracts/Profile.sol:140]: Done

File: contracts/storage/ProfileStorage.sol (hunk 3 above)
Comment (u-hubar) [reply contracts/storage/ProfileStorage.sol:94]: Changed

**Review by branarakic — APPROVED** [review]

**Pull request merged**